src/renderers/webgl/WebGLRenderLists.js

```diff
@@ -12,6 +12,10 @@ function painterSortStable( a, b ) {
 
 		return a.renderOrder - b.renderOrder;
 
+	} else if ( a.materialOrder !== b.materialOrder ) {
```

Comment:
Unsure whether the materialOrder should come after Group / Object's renderOrder or not